meilenstein_02/analyse_Speichermedium/README.md

```diff
@@ -0,0 +1,103 @@
+<!-- Alle relevanten Sachen aufschreiben, später kann aussortiert werden! -->
+    - Speicher/Schreiben
+    - Lesen/Suchen
+    - Löschen/Aendern
+    - Sortieren
+
+## Liste der Klassen:
+<!-- Erstmal alle aufschreiben, dann auswählen! (Kriterien siehe Vorgehensweise) -->
+<!-- Warum sind die Klassen existent? Wenn das zu beantworten ist - u good! -->
+<!-- ausgewählte Klassen mit Link, andere einklammern und CRC-Karte löschen -->
+- [Suchen](crc-{Suchen}.md)
+- [Speichern](crc-{Speichern}.md)
+- [Aendern](crc-{Aendern}.md)
+- (Sortieren)
+
+---
+
+# Analyse - Speichermedium
+<!-- Hier Notizen zum Denkprozess! -->
+1. Welche Aufgaben hat das Speichermedium/soll es zur Verfügung stellen?
+     - ***Es muss zuerst die Art des Speichermediums entschieden werden***.. (Datenbank oder Datei) ..
+     ***Da dies noch in Besprechung ist, gehe ich aktuell von eine Datei aus.***
+     - Neben dem **Als was** etwas abgespeichert werden soll gibt es noch andere Anforderungen
+     - Es existiert eine Abstrakte Oberklasse **Speichefunktionen* die in Verbindungen mit den Funktionen an sich steht
+     - Die Funktionen die zum aktuellen Zeitpunkt zur Verfügung gestellt werden sollen, sind **Speichern/Schreiben**, **Suchen/Lesen** und **Ändern**
+     - Damit ist gemeint, dass die Funktion **Suchen** dem *DataClient* zur Verfügung stehen soll und die Funktionen **Speichern** und **Löschen/Aendern**, den Funktionen *Zoom Maps*, *Codecharts*, *Webcam basiertes Eyetracking* und *ConfigFile*
+     - Da die 3 großen Funktionen aus mehreren Unterfunktione besteht, werden diese zu jeweils 3 Klassen
+2. Was macht die Klasse ***Suchen*** ?
+     - Diese muss nur dem DataClient zur Verfügung gestellt werden
+     - DataClient kann speziell nach Informationen suchen
+     - Besitzt für alle 3 Funktionen spezielle Suchen, für die jeweiligen Daten
+     - Besitzt die Möglichkeit nach Dateinnamen als auch nach allen Daten von einer Funktion zu suchen
+     - Kann einzelne Daten (falls notwendig) als auch einen kompletten Datensatz ausgeben
+3. Was macht die Klasse ***Speichern***
+     - Diese Klasse steht den drei Funktionen, sowie der ConfigFile zur Verfügung
+     - Es wird für jede Funktion eine eigene Datei erstellt in der die Informationen gespeichert werden, dadurch wird verhindert das es zu "Vermischungen" der Daten kommt
+     - durch das getrennte Speichern wird auch die Klasse *Suchen* leichter zu implementieren
+     - Stellt jeder Funktion eine einheitliche Speicherung der Infromationen in der Datei sicher
+     - muss in Absprache mit den Funktionen erstellt werden
+     - besitzt die Möglichkeit den Dateien Namen zu geben, ansonsten standard -> Datum der Erstellung
+4. Was macht die Klasse ***Löschen/Aendern***
+     - Sie ermöglicht es alte Datensätze zu löschen
+     - Steht dem Dataclient zur Verfügung (Falls gewünscht ist, das nach der Analyse die Daten gelöscht werden sollen (oder weil nur die graphische Information von interesse ist))
+     
```

Comment:
```suggestion
     Ich würde für die Übersicht die Punkte bei Verantwortlichkeiten und Kollaborationen anders formatieren, rich diff ist momentan sehr unübersichtlich
```

Reply:
Die eigentlich Sachen werden ja alle in `Verantwortlichkeiten` und `Kollaborationen` schon kurz und knackig behandelt, vielleicht könnte man diese Abschnitte in die CRC-Karten verschieben? In den dortigen Notizabschnitt?

Reply:
Hmmm ich finde das halt als schöne Dokumentation zusammengefasst, wie ich angefangen habe bis zum Ende und an sich steht das auch alles nochmal in den CRC Karten

Reply:
Ah okay :)
Da hätte ich noch die Idee, den Abschnitt weiter hinter zu verschieben, sodass du ganz am Ende das hier hinzufügst:
```md
---
## Notizen
... und hier der Inhalt ...
```

Reply:
Dann steht doch aber bei der Analyse nichts mehr ?

Reply:
Da hab ich immer und Philip auch, einen kleinen Überblick über die Funktionsweise des Moduls insgesamt reingeschrieben (siehe #25, #23, #22)